src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -74,6 +74,14 @@ private void HandleBinaryOperator(in OperationAnalysisContext context, INamedTyp
                 return;
             }
 
+            if (binary.Language == LanguageNames.VisualBasic)
+            {
+                if (IsSymbolClassType(binary.LeftOperand) || IsSymbolClassType(binary.RightOperand))
```

Comment:
Curious: why are we bailing out for reference typed operands for VB?

Reply:
The compiler code use `x Is y` as the VB equivalent of `(object)x == y`.